ffcx/codegeneration/lnodes.py

```diff
@@ -720,31 +722,55 @@ def as_statement(node):
             return as_statement(node[0])
         else:
             return StatementList(node)
+    elif isinstance(node, Section):
+        return node
     else:
         raise RuntimeError(
             "Unexpected Statement type %s:\n%s" % (type(node), str(node))
         )
 
 
+class Section(LNode):
+    """A section of code with a name and a list of statements."""
+
+    def __init__(self, name: str, statements: List, input: Optional[List] = None,
+                 output: Optional[List] = None, annotations: Optional[List] = None):
+        self.name = name
+        self.statements = [as_statement(st) for st in statements]
+        self.annotations = annotations or []
+        self.input = input or []
+        self.output = output or []
+
+        # Convert all inputs to strings
```

Comment:
Can we keep `input` and `ouput` as `Symbols` - so we can see the type, and maybe dimension?